src/components/PetEditPage.jsx

```diff
@@ -0,0 +1,167 @@
+                            ))}
+                        </select>
+                    </div>
+
+                    {/* Only show form if a pet is selected */}
+                    {selectedPetIndex !== null && (
+                        <form onSubmit={handleSubmit}>
+                            <div>
+                                <label htmlFor="petName">Name:*</label>
+                                <input type="text" id="petName" name="petName" value={formData.petName} onChange={handleChange} required />
+                            </div>
+                            <div>
+                                <label htmlFor="petBreed">Breed:*</label>
+                                <input type="text" id="petBreed" name="petBreed" value={formData.petBreed} onChange={handleChange} required />
+                            </div>
+                            <div>
+                                <label htmlFor="petSex">Sex:*</label>
+                                <select id="petSex" name="petSex" value={formData.petSex} onChange={handleChange} required>
+                                    <option value="">Select Sex</option>
+                                    <option value="Male">Male</option>
+                                    <option value="Female">Female</option>
+                                </select>
+                            </div>
+                            <div>
+                                <label htmlFor="petBirthDate">Birth Date:*</label>
+                                <input type="date" id="petBirthDate" name="petBirthDate" value={formData.petBirthDate} onChange={handleChange} required />
+                            </div>
+                            <div>
+                                <label htmlFor="lastCheckUp">Last Check-Up Date:</label>
+                                <input type="date" id="lastCheckUp" name="lastCheckUp" value={formData.lastCheckUp} onChange={handleChange} />
+                            </div>
+                            <div>
+                                <label htmlFor="nextCheckUp">Next Check-Up Date:</label>
+                                <input type="date" id="nextCheckUp" name="nextCheckUp" value={formData.nextCheckUp} onChange={handleChange} />
+                            </div>
+                            <div>
+                                <label htmlFor="specialCondition">Special Condition:</label>
+                                <textarea id="specialCondition" name="specialCondition" value={formData.specialCondition} onChange={handleChange} rows="4" cols="50" placeholder="Enter any special conditions or notes about the pet..." />
+                            </div>
+                            <button type="submit" className="formButton">Edit</button>
+                        </form>
+                    )}
+                </div>
+                <ToastContainer
+                    autoClose={1000}
+                />
```

Comment:
We probably need this component at the component level, it should be globally available.